src/Razor/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentChildContentIntegrationTest.cs

```diff
@@ -115,25 +115,38 @@ public void ChildContent_ExplicitChildContent_UnrecogizedContent_ProducesDiagnos
                 diagnostic.GetMessage(CultureInfo.CurrentCulture));
         }
 
-        [Fact]
-        public void ChildContent_ExplicitChildContent_UnrecogizedElement_ProducesDiagnostic()
+        [Theory]
+        [InlineData("UnrecognizedChildContent", true, true)]
+        [InlineData("UnrecognizedChildContent", false, true)]
+        [InlineData("繁体字", true, true)]
+        [InlineData("繁体字", false, false)]
+        public void ChildContent_ExplicitChildContent_UnrecogizedElement_ProducesDiagnostic(string unrecognizedComponentName, bool supportLocalizedComponentNames, bool diagnosticExpected)
```

Comment:
Could we make these two separate tests? `ChildContent_ExplicitChildContent_StartsWithCharThatIsOtherLetterCategory_WhenLocalizedComponentNamesIsAllowed` / `ChildContent_ExplicitChildContent_StartsWithCharThatIsOtherLetterCategory_WhenLocalizedComponentNamesIsDisalowed`